src/axom/slam/mesh_struct/IA.hpp

```diff
@@ -0,0 +1,402 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * \file IA.hpp
+ *
+ * \brief Contains the header information of IA class
+ */
+
+#ifndef SLAM_IA_H_
+#define SLAM_IA_H_
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+
+#include "axom/slam/policies/IndirectionPolicies.hpp"
+#include "axom/slam/policies/CardinalityPolicies.hpp"
+
+#include "axom/slam/Set.hpp"
+#include "axom/slam/RangeSet.hpp"
+#include "axom/slam/DynamicSet.hpp"
+#include "axom/slam/StaticRelation.hpp"
+#include "axom/slam/DynamicConstantRelation.hpp"
+
+#include "axom/slam/Utilities.hpp"
+#include "axom/slam/Map.hpp"
+#include "axom/slam/DynamicMap.hpp"
+#include "axom/slam/FieldRegistry.hpp"
+
+namespace axom
+{
+namespace slam
+{
+/**
+ * \class IA
+ *
+ * \brief Implements the Indexed mesh data structure with Adjacencies (IA)
+ *
+ * \tparam TDIM The topological dimension (2=triangle, 3=tetrahedra).
+ * \tparam SDIM The dimension of the embedding space.
+ * \tparam PointType A point type for the mesh position data
+ *
+ * \details The IAMesh class is an adjacency-based topological mesh data
+ * structure for simplicial complexes. It encodes:
+ * - the boundary relation from elements to their vertices
+ * - the partial coboundary relation from vertices to one incident element, and
+ * - the adjacency relation between elements along their facets (faces of dimension TDIM-1).
+ *
+ * PointType is required to have the following interface:
+ * - .ctor(T*)          -- constructor from an array of T
+ * - operator[](index)  -- subscript operator to access the coordinates
+ * - data() -> T*       -- direct access to the data
+ * - operator==()       -- equality operator
+ */
```

Comment:
Is this a well-known data structure?  Please cite the paper (or web page, even).